server/src/languageServerBase.ts

```diff
@@ -208,6 +226,50 @@ export abstract class LanguageServerBase implements LanguageServerInterface {
         });
     }
 
+    createFileWatcher(paths: string[], listener: FileWatcherEventHandler): FileWatcher {
+        // Capture "this" so we can reference it within the "close" method below.
+        const lsBase = this;
+
+        // Determine which paths are located within one or more workspaces.
+        // Those are already covered by existing file watchers handled by
+        // the client.
+        const workspacePaths: string[] = [];
+        const nonWorkspacePaths: string[] = [];
+        const workspaces = this._workspaceMap.getNonDefaultWorkspaces();
+
+        paths.forEach(path => {
+            if (workspaces.some(workspace => containsPath(workspace.rootPath, path))) {
+                workspacePaths.push(path);
+            } else {
+                nonWorkspacePaths.push(path);
+            }
+        });
+
+        // For any non-workspace paths, use the node file watcher.
+        const nodeWatchers = nonWorkspacePaths.map(path => {
+            return fs.watch(path, { recursive: true }, listener);
```

Comment:
This bypasses the VFS layer, which doesn't seem right from the abstraction point of view, as I assume we're going to want to test this.

Reply:
This isn't any different from the current implementation, which no-ops file watchers in the mocked file system implementation.

Reply:
I don't agree that it isn't different; now this file imports `fs` directly whereas before it had to do so via the VFS interface. The only file that should know how to access the file system for real should be the "real" implementation of the VFS interface, but now both the VFS and the language server does it.